ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -785,9 +782,12 @@ export const RuntimePanel = fp.flow(
   ])([currentRuntime, status]);
   const [panelContent, setPanelContent] = useState<PanelContent>(initialPanelContent);
 
-  const [selectedDiskSize, setSelectedDiskSize] = useState(diskSize);
   const [selectedMachine, setSelectedMachine] = useState(initialMasterMachine);
+  const [selectedDiskSize, setSelectedDiskSize] = useState(diskSize);
   const [selectedCompute, setSelectedCompute] = useState<ComputeType>(initialCompute);
+  // Note: while the Dataproc config does contain masterMachineType and masterDiskSize,
```

Comment:
maybe there's some kind of refactoring we can do to reduce the data duplication in these objects (dataproc.masterMachineType can be a function that returns a value from the parent?) but I think this looks good for now

Reply:
I think a way to fix this would be to have either a partial DataprocConfig type which only covered the worker settings. We can't directly use the DataprocConfig type for this, since that's tied to the API model. Probably the thing that's missing here is either using the API model throughout (keeping the UI elements somewhat independent of the model), or using alternate leaner models that more directly map the UI, but need to be converted at the boundary to API models.